.github/workflows/k8s-apply.yaml

```
@@ -113,7 +128,7 @@ jobs:
   delete_runner:
     name: Delete Self-Hosted Runner
     needs: [create_runner, deploy]
-    if: ${{ always() }}
+    if: ${{ inputs.run_k8s_workflow == true }}
```

```suggestion
    if: ${{ always() && inputs.run_k8s_workflow == true }}
```
I know this looks strange, but it's a workaround to force this job to run in the case when
- `run_k8s_workflow == true`
- the `deploy` step fails for some reason
otherwise it would not delete the runners